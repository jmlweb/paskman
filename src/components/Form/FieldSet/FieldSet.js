import styled from 'styled-components';
import spacing from '../../../styles/spacing';

const FieldSet = styled.fieldset`
  border: 0;
  margin: 0 0 ${spacing.sm};
  padding: 0;
`;

export default FieldSet;
