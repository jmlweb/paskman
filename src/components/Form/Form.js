// @flow
import styled from 'styled-components';
import spacing from '../../styles/spacing';
import type { Target } from 'styled-components';

const Form: Target = styled.form`
  margin: 0;
  padding: ${props => props.noSpacing ? 0 : `${spacing.md} ${spacing.sm}`};
`;

export default Form;
