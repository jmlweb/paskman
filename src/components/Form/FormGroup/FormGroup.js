// @flow
import styled from 'styled-components';
import spacing from '../../../styles/spacing';

const FormGroup = styled.div`
  margin-bottom: ${spacing.sm};
`;

export default FormGroup;
