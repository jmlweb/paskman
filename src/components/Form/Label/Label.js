// @flow
import styled from 'styled-components';
import colors from '../../../styles/colors';
import type { Target } from 'styled-components';

const Label: Target = styled.label`
  font-weight: 500;
  display: block;
  font-size: 1.4rem;
  text-transform: uppercase;
  margin-bottom: 0.5rem;
  color: ${colors.secondaryA7};
  b,
  strong {
    font-weight: 700;
  }
`;

export default Label;
