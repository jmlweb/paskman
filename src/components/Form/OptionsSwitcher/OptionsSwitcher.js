import React from 'react';
import PT from 'prop-types';
import styled, { css } from 'styled-components';
import colors from '../../../styles/colors';
import spacing from '../../../styles/spacing';
import timings from '../../../styles/timings';

const optionsSwitcherBorder = `2px solid ${colors.secondaryT7}`;

const StyledOptionsSwitcher = styled.div`
  display: flex;
  border-top: ${optionsSwitcherBorder};
  border-bottom: ${optionsSwitcherBorder};
  border-left: ${optionsSwitcherBorder};
  border-radius: 4px;
  overflow: hidden;
`;

const selectedItem = css`
  background: ${colors.secondaryT7};
  color: ${colors.secondary};
`;

const Item = styled.label`
  flex: 1;
  text-align: center;
  background: ${colors.secondaryA1};
  padding: ${spacing.xs};
  border-right: 2px solid ${colors.secondaryT7};
  transition: background 0.2s ${timings.easeOutSine};
  font-size: 1.5rem;
  font-weight: 700;
  color: ${colors.secondaryA7};
  ${props => props.selected && selectedItem};
`;

const Input = styled.input.attrs({
  type: 'radio',
})`
  position: absolute;
  opacity: 0;
`;

const OptionsSwitcher = ({ options, value, onChange }) => (
  <StyledOptionsSwitcher>
    {options.map(
      option => (
        <Item
          selected={option.value === value}
          key={+option.value}
        >
          <Input
            id={`id-${+option.value}`}
            value={option.value}
            onClick={onChange}
          />
          <span>{option.description}</span>
        </Item>
      )
    )}
  </StyledOptionsSwitcher>
);

OptionsSwitcher.propTypes = {
  options: PT.arrayOf(PT.object).isRequired,
  value: PT.oneOfType([PT.string, PT.number, PT.bool]).isRequired,
  onChange: PT.func.isRequired,
};

export default OptionsSwitcher;
