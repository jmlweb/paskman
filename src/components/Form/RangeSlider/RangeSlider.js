import React from 'react';
import styled, { css } from 'styled-components';
import Slider from 'react-rangeslider';
import colors from '../../../styles/colors';

const fillStyle = css`
  display: block;
  box-shadow: inset 0 1px 3px ${colors.secondaryA2};
`;

const StyledRangeSlider = styled(Slider)`
  margin: 10px 0 25px;
  position: relative;
  background: ${colors.secondaryT7};
  touch-action: none;
  ${fillStyle};
  .rangeslider__fill {
    ${fillStyle};
  }
  .rangeslider__handle {
    background: ${colors.secondary};
    border: 1px solid transparent;
    cursor: pointer;
    display: inline-block;
    position: absolute;
    .rangeslider__tooltip {
      width: 80px;
      height: 40px;
      text-align: center;
      position: absolute;
      background-color: ${colors.blackA8};
      font-weight: normal;
      font-size: 14px;
      transition: all 100ms ease-in;
      border-radius: 4px;
      display: inline-block;
      color: white;
      opacity: 0;
      &:hover & {
        opacity: 1;
      }
      &:after {
        content: ' ';
        position: absolute;
        width: 0;
        height: 0;
      }
      span {
        margin-top: 12px;
        display: inline-block;
        line-height: 100%;
      }
    }
  }
  &.rangeslider-horizontal {
    height: 12px;
    border-radius: 10px;
    .rangeslider__fill {
      height: 100%;
      background-color: ${colors.secondaryT7};
      border-radius: 10px;
      top: 0;
    }
    .rangeslider__handle {
      width: 20px;
      height: 20px;
      border-radius: 100%;
      top: -6px;
    }
    .rangeslider__tooltip {
      top: -55px;
      left: -27px;
      &:after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        border-top: 8px solid ${colors.blackA8};
        left: 30px;
        bottom: -8px;
      }
    }
    .rangeslider__label-list .rangeslider__label {
      position: absolute;
      font-size: 14px;
      cursor: pointer;
      display: inline-block;
      top: 10px;
    }
  }
`;

const RangeSlider = props => <StyledRangeSlider {...props} />;

export default RangeSlider;
