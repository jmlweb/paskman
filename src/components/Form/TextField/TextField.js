import styled from 'styled-components';
import colors from '../../../styles/colors';
import fonts from '../../../styles/fonts';

const TextField = styled.input`
  border: 2px solid ${colors.secondaryA4};
  border-radius: 4px;
  box-sizing: border-box;
  display: block;
  font-family: ${fonts.fontFamily};
  font-size: 1.7rem;
  padding: 8px 12px;
  width: 100%;
  &:focus {
    border: 2px solid ${colors.secondary};
    outline: 0;
  }
`;

export default TextField;
