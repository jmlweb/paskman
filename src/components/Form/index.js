export { default as Form } from './Form';
export { default as FieldSet } from './FieldSet/FieldSet';
export { default as FormGroup } from './FormGroup/FormGroup';
export { default as Label } from './Label/Label';
export { default as TextField } from './TextField/TextField';
export { default as OptionsSwitcher } from './OptionsSwitcher/OptionsSwitcher';
export { default as RangeSlider } from './RangeSlider/RangeSlider';
