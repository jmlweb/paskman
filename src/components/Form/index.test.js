import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { default as Form } from './Form';
import { default as FieldSet } from './FieldSet/FieldSet';
import { default as FormGroup } from './FormGroup/FormGroup';
import { default as Label } from './Label/Label';
import { default as TextField } from './TextField/TextField';
import { default as OptionsSwitcher } from './OptionsSwitcher/OptionsSwitcher';
import { default as RangeSlider } from './RangeSlider/RangeSlider';

configure({ adapter: new Adapter() });

describe('Button', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <Form>
        <FieldSet>
          <FormGroup>
            <Label>Dummy</Label>
            <TextField />
            <OptionsSwitcher
              options={[{description: 'dummy', value: 1}]}
              value={1}
              onChange={jest.fn()}
            />
            <RangeSlider
              tooltip={false}
              min={1}
              max={10}
              step={1}
              value={5}
              onChange={jest.fn()}
            />
          </FormGroup>
        </FieldSet>
      </Form>
    );
  });
  it('Render the component', () => {
    expect(wrapper.find(Form).length).toEqual(1);
  });
  it('Matchs snapshot', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
