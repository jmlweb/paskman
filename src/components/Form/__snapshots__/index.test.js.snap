// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Matchs snapshot 1`] = `
<form
  className="sc-bdVaJa fAoWzh"
>
  <fieldset
    className="sc-bwzfXH gSXRtU"
  >
    <div
      className="sc-htpNat QOKog"
    >
      <label
        className="sc-bxivhb fMASpP"
      >
        Dummy
      </label>
      <input
        className="sc-ifAKCX fPNPKI"
      />
      <div
        className="sc-EHOje erXxej"
      >
        <label
          className="sc-bZQynM bXIApl"
          selected={true}
        >
          <input
            className="sc-gzVnrw bLHfQE"
            id="id-1"
            onClick={[Function]}
            type="radio"
            value={1}
          />
          <span>
            dummy
          </span>
        </label>
      </div>
      <div
        aria-orientation="horizontal"
        aria-valuemax={10}
        aria-valuemin={1}
        aria-valuenow={5}
        className="rangeslider rangeslider-horizontal sc-htoDjs gqLgCW"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="rangeslider__fill"
          style={
            Object {
              "width": "0px",
            }
          }
        />
        <div
          className="rangeslider__handle"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          style={
            Object {
              "left": "0px",
            }
          }
          tabIndex={0}
        >
          <div
            className="rangeslider__handle-label"
          >
            
          </div>
        </div>
        <ul
          className="rangeslider__labels"
        />
      </div>
    </div>
  </fieldset>
</form>
`;
